### ISSUE TITLE
BUG - Sev 3 - iOS - Link - OS font scaling grows Icon, but does not shrink it

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/va-mobile-library#203]
## What happened?

While working on #168 it was noted that font scaling has a slight delay in scaling up the icon to match the bigger text. Not terribly problematic. However, when scaling the font down, the icon remains stuck much larger than it ought to be. Only tested in Storybook (not app). Navigating to a different story and back correctly rescales it. It is believed to maybe be an issue around rerendering, hence the upscaling working correctly but with a slight delay relative to the text.

This was not observed on Android which insisted on completely rebuilding the Expo App/Storybook whenever font size was changed. The icon scales up and down fine in iOS for all non-inline variants.

Video of bugged behavior:

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/8690976/aff07892-9cd3-415c-af36-76738474ebcd

## Steps to Reproduce

1. Navigate to Storybook in iOS
2. Open a non-Inline variant of Link
3. Scale device text size up
4. **Note**: Both icon and text grow as expected
5. Scale device text size back down
6. **Note**: Both icon and text shrink as expected
7. Open the Inline variant of Link
8. Scale device text size up
9. **Note**: Both icon and text grow as expected (icon may exhibit have a slight delay in scaling if you get back fast enough)
10. Scale device text size back down
11. **Note**: Text shrinks as expected, but the icon remains larger than it should be until navigating to another story then back

## Desired behavior

Icon should scale back down appropriately with the text.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High
- Frequency: Low 

- 2 - High

## Linked to Story

Noted during #168

**Comment by TimRoe** [comment]
With the decision to remove inline Links for now, this bug is no longer present. However, slotted into the Paragraph component epic as it may return when trying to implement similar again--and, if not, can be closed then.

**Comment by bischoffa** [comment]
Unable to add severity label to this ticket. @TimRoe can you please add it

**Comment by TimRoe** [comment]
Added.

**Comment by kellylein** [comment]
Holding on this as it is tied to component(s) we haven't built yet.
